pkg/services/publicdashboards/service/service.go

```diff
@@ -238,6 +245,79 @@ func (pd *PublicDashboardServiceImpl) GetMetricRequest(ctx context.Context, dash
 	return metricReqDTO, nil
 }
 
+func (pd *PublicDashboardServiceImpl) GetAnnotations(ctx context.Context, reqDTO AnnotationsQueryDTO, accessToken string) ([]*annotations.ItemDTO, error) {
+	_, dash, err := pd.GetPublicDashboard(ctx, accessToken)
+	if err != nil {
+		return nil, err
+	}
+
+	type annotation struct {
+		Datasource struct {
+			DType string `json:"type"`
+			Uid   string `json:"uid"`
+		}
+		Enable bool   `json:"enable"`
+		Type   string `json:"type"`
+		Target struct {
+			Limit    int64    `json:"limit"`
+			MatchAny bool     `json:"matchAny"`
+			Tags     []string `json:"tags"`
+			Type     string   `json:"type"`
+		}
+	}
+
+	type annotationsDto struct {
+		Annotations struct {
+			List []annotation `json:"list"`
+		}
+	}
+
+	dto := annotationsDto{}
+	bytes, err := dash.Data.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(bytes, &dto)
+	if err != nil {
+		return nil, err
+	}
+
+	anonymousUser, err := pd.BuildAnonymousUser(ctx, dash)
+	if err != nil {
+		return nil, err
+	}
+
+	var results []*annotations.ItemDTO
+	for _, anno := range dto.Annotations.List {
+		if anno.Enable {
```

Comment:
Because this is only dealing with Grafana's built-in annotation data source here (and many use Prometheus, loki or Elasticsearch instead to query for annotations)   you need to check the annotation query datasource prop. 

Check if the uid is grafanads.DatasourceUID

Reply:
Yep, this is still a work in progress. Just put up the draft PR for visibility